grammar/stmt.g

```diff
@@ -0,0 +1,7 @@
+Stmt =
+    Item:Item |
+    Local:{ attrs:OuterAttr* "let" pat:Pat { ":" ty:Type }? { "=" init:Expr }? ";" } |
+    Expr:Expr |
+    Semi:";";
```

Comment:
If nonterminals in this grammar are supposed to match fragments in `macro_rules`, then `Stmt` needs to be organized slightly differently, more like

```
Stmt =
    Item:Item |
    Local:{ attrs:OuterAttr* "let" pat:Pat { ":" ty:Type }? { "=" init:Expr }? } |
    Expr:Expr;

StmtX =
    Item |
    Local ";" |
    Expr |
    ";"
 Block = "{" attrs:InnerAttr* StmtX* "}";
```